Source/Checker/Checker.cpp

```diff
@@ -25,6 +25,32 @@
 // Helpers
 //---------------------------------------------------------------------------
 
+
+//---------------------------------------------------------------------------
+std::string xml_encode(const std::string& data)
+{
+    std::stringstream ss;
+    for (size_t pos=0; pos<data.size(); pos++)
+    {
+        switch (data[pos])
+        {
+            case '\'': ss << "&apos;"; break;
+            case '"': ss << "&quot;"; break;
+            case '&': ss << "&amp;"; break;
+            case '<': ss << "&lt;"; break;
+            case '>': ss << "&gt;"; break;
+            case '\n': ss << "&#xA;"; break;
+            case '\r':
+                ss << "&#xA;";
+                if (pos+1<data.size() && data[pos+1]=='\n') // translate the #xD #xA sequence to a single #xA character
+                    pos++;
+            break;
+            default: if ((unsigned char)data[pos]>=0x20) ss << data[pos]; // Ignore others control characters
```

Comment:
From XML spec:
`  Char = #x9 | #xA | #xD | [#x20- [...]`
--> `\t` should not be ignored.